### PR TITLE
Bugfixing and improvements for Blockchain/WaletService

### PR DESCRIPTION
**Pull request by mnaichuk**

### CODE REVIEW

File: app/services/blockchain_service2.rb [hunk 1]
```diff
@@ -58,7 +64,7 @@ def filter_withdrawals(block)
     block.select { |transaction| transaction.hash.in?(withdraw_txids) }
   end
 
-  def update_or_create_deposit(transaction)
+  def update_or_create_deposit(transaction, deposits = [])
```
Comment (mnaichuk) [inline app/services/blockchain_service2.rb]: remove second param

File: app/services/blockchain_service2.rb [hunk 2]
```diff
@@ -68,6 +74,9 @@ def update_or_create_deposit(transaction)
       return
     end
 
+    # TODO: Rewrite this guard clause
+    return unless PaymentAddress.find_by(currency_id: transaction.currency_id, address: transaction.to_address)
```
Comment (mnaichuk) [inline app/services/blockchain_service2.rb]: use exist? instead

File: lib/peatio/bitcoin/client.rb [hunk 3]
```diff
@@ -29,6 +29,8 @@ def json_rpc(method, params = [])
       response = JSON.parse(response.body)
       response['error'].tap { |error| raise ResponseError.new(error['code'], error['message']) if error }
       response.fetch('result')
+    rescue Faraday::Error => e
```
Comment (mnaichuk) [inline lib/peatio/bitcoin/client.rb:33]: Add ConnectionError

File: lib/peatio/ethereum1/wallet.rb [hunk 4]
```diff
@@ -50,7 +50,7 @@ def prepare_deposit_collection!(transaction, deposit_spread, deposit_currency)
 
       # We collect fees depending on the number of spread deposit size
       # Example: if deposit spreads on three wallets need to collect eth fee for 3 transactions
-      fees = convert_from_base_unit(options.fetch(:gas_limit) * options.fetch(:gas_price))
+      fees = convert_from_base_unit(options.fetch(:gas_limit).to_i * options.fetch(:gas_price).to_i)
```
Comment (mnaichuk) [inline lib/peatio/ethereum1/wallet.rb:53]: coerce to integer in configure step

**Review by ysv — APPROVED** [review]

**Pull request merged**